src/main/java/pl/shopgen/controllers/Search.java

```diff
@@ -0,0 +1,49 @@
+package pl.shopgen.controllers;
+
+import javafx.util.converter.BigDecimalStringConverter;
+import org.springframework.web.bind.annotation.*;
+import pl.shopgen.models.CategoryRepository;
+import pl.shopgen.models.Product;
+import pl.shopgen.models.ProductRepository;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/search")
+public class Search {
+    private final ProductController productController;
+
+    public Search(CategoryRepository categoryRepository, ProductRepository productRepository, ProductController productController) {
+        this.productController = productController;
+    }
+
+    @RequestMapping(value = {"", " "}, method = RequestMethod.GET)
+    @ResponseBody
+    List<Product> findByName(@RequestParam Map<String,String> allRequestParams) {
+        List<Product> productList = new ArrayList<Product>();
+        String  productName = allRequestParams.getOrDefault("name", null);
+        String  categoryName =allRequestParams.getOrDefault("category",null);
+        String lowerPriceString = allRequestParams.getOrDefault("lowerPrice","0");
+        String  higherPriceString =allRequestParams.getOrDefault("higherPrice","10000000");
+        BigDecimal higherPrice=new BigDecimal(higherPriceString);
+        BigDecimal lowerPrice=new BigDecimal(lowerPriceString);
+        Predicate<Product> nameFilter = p ->   productName == null ? true :  p.getName().equalsIgnoreCase(productName);
+        Predicate<Product> categoryFilter = p ->   categoryName == null ? true :  p.getCategory().getName().equalsIgnoreCase(categoryName);
```

Comment:
id kategorii, nie nazwa

Reply:
Jak kategoria nie ma id?? Znaczy  ma ale jakieś takie 5af0aba9b9bb5d1ab08e7782??

Reply:
To jest to id, nie szukamy po nazwie bo równocześnie znajdą się spodnie damskie i męskie

Reply:
A jak frontend będzie to przesyłał? Mogę poprawić ale wtedy musieliby przesłać coś takiego /category=5af0aba9b9bb5d1ab08e7782;

Reply:
Tak zrobią, będą gdzieś u siebie trzymać idki

Reply:
Ok poprawione